lib/ts/recipe/thirdparty/recipe.ts

```diff
@@ -76,6 +78,160 @@ export default class Recipe extends RecipeModule {
+                                if (!emailsResult.includes(orderedLoginMethodsByTimeJoinedOldestFirst[i].email!)) {
+                                    emailsResult.push(orderedLoginMethodsByTimeJoinedOldestFirst[i].email!);
+                                }
+                            }
+                        }
+                        return {
+                            status: "OK",
+                            factorIdToEmailsMap: {
+                                thirdparty: emailsResult,
+                            },
+                        };
+                    } else if (nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined.length === 1) {
+                        // we return just this email cause if this thirdparty
+                        // user is from the same tenant that the user is logging into,
+                        // then they have to use this since new factor setup won't
+                        // be allowed. Even if this thirdparty user is not from
+                        // the same tenant as the user's session, we still return just
+                        // this cause that way we still just have one thirdparty
+                        // loginMethod for this user, just across different tenants.
+                        return {
+                            status: "OK",
+                            factorIdToEmailsMap: {
+                                thirdparty: nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined,
+                            },
+                        };
+                    }
+
+                    // Finally, we return all emails that have thirdparty login
+                    // method for this user, but keep the session's email first
+                    // if the session's email is in the list of
+                    // nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined (for better UX)
+                    let emailsResult: string[] = [];
+                    if (
+                        sessionLoginMethod.email !== undefined &&
+                        nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined.includes(sessionLoginMethod.email)
+                    ) {
+                        emailsResult = [sessionLoginMethod.email];
+                    }
+
+                    for (let i = 0; i < nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined.length; i++) {
+                        if (!emailsResult.includes(nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined[i])) {
+                            emailsResult.push(nonFakeEmailsThatHaveThirdPartyLoginMethodOrderedByTimeJoined[i]);
+                        }
+                    }
+
+                    return {
+                        status: "OK",
+                        factorIdToEmailsMap: {
+                            thirdparty: emailsResult,
+                        },
+                    };
+                });
+            }
```

Comment:
this needs to be rethought: https://supertokens.slack.com/archives/C051BN9QJUX/p1705166104069829